fs/copy.go

```diff
@@ -9,6 +9,16 @@ import (
 	"github.com/pkg/errors"
 )
 
+// Mode indicates whether to use hardlink or copy content
+type Mode int
+
+const (
+	// Content creates a new file, and copies the content of the file
+	Content Mode = iota
+	// Hardlink creates a new hardlink to the existing file
+	Hardlink
```

Comment:
Where is this used other than in one of moby's graph drivers? Moby specific changes should be kept in that repo, this project is focused on cleaner interfaces to support containerd and future fs metadata formats.